src/workingtitle-vcockpits-instruments-navsystems/html_ui/Pages/VCockpit/Instruments/NavSystems/AS1000/PFD/AS1000_PFD.js

```diff
@@ -33,19 +33,65 @@ class AS1000_PFD extends BaseAS1000 {
+
+    async pfdConfig() {
+        this.loadSavedBrightness("PFD");
+        this.loadSavedBrightness("MFD");
+        // It seems to be the case that in some planes the avionics brightness knob value changes
+        // as the plane is in the process of intializing -- the stock Bonanza seems to be one of
+        // these.  This causes the code in onUpdate that determines if the knob has moved from its
+        // prior value to trigger, meaning that the brightness that we set is immediately
+        // overridden by the new position of the knob.  The only way I have found to set this is,
+        // unfortunately, to stick a little pause in here to let the systems load a bit before
+        // completing the initialization of our current settings.  It's kinda gross, but since this
+        // is an async function it's not *too* bad, and the worst it means it that the hardware
+        // knob, if present, won't function for a few seconds.
+        //
+        // The 10000ms value here was experimentally derived by testing to see what caused loads in
+        // troublesome planes to succeed on my system.  It may need adjustment if other folks find
+        // themselves having that issue. 
+        await new Promise(resolve => setTimeout(resolve, 10000));
+        // We need to wait for this to finish before we can do the initial set of the light pot
+        // in the line below because this can set a custom value for the avionics knob.
+        await this._cfgHandler.loadModelFile("interior").then((dom) => { this.processInteriorConfig(dom) });
+        this.avionicsKnobValue = SimVar.GetSimVarValue("A:LIGHT POTENTIOMETER:" + this.avionicsKnobIndex, "number") * 100;
+        this._pfdConfigDone = true;
+    }
+
+    processInteriorConfig(dom) {
+        let templates = dom.getElementsByTagName("UseTemplate");
+        for (const item of templates) {
+            if (item.getAttribute("Name").toLowerCase() != "asobo_as1000_pfd_template")
+                continue;
+            let children = item.childNodes;
+            for (const item of children) {
+                if (item.nodeName.toLowerCase() != "potentiometer")
+                    continue;
+                this.avionicsKnobIndex = item.textContent
+            }
         }
-        this.avionicsKnobValue = avionicsKnobValueNow
     }
+
+    onUpdate(_deltaTime) {
+        if (this._pfdConfigDone) {
+            let avionicsKnobValueNow = SimVar.GetSimVarValue("A:LIGHT POTENTIOMETER:" + this.avionicsKnobIndex, "number") * 100;
+            if (avionicsKnobValueNow != this.avionicsKnobValue) {
+                this.setBrightness("PFD", avionicsKnobValueNow);
+                this.setBrightness("MFD", avionicsKnobValueNow);
+                this.avionicsKnobValue = avionicsKnobValueNow;
+            }
+        } 
+    }
+
     onEvent(_event) {
         if (_event == "MENU_Push") {
+            // Uncomment this for easy debugging of config issues.
+            // this.pfdConfig();
```

Comment:
You can use
`g_modDebugMgr.AddDebugButton("InsertTitleHere", function() {dosomething});`

This will always show a button in the devkit panel that can do stuff

Reply:
Ah, sweet.  I remember that from the first time I read the devkit docs but I'd never used it then forgot.  :D  Thanks for the reminder.